.github/workflows/pr.yml

```
@@ -19,8 +19,20 @@ jobs:
     # One runner for each domain
     strategy:
       matrix:
-        domain: [blas, dft, lapack, rng]
-    name: MKL ${{ matrix.domain }} CPU
+        include:
+        - config: portBLAS
+          options: -DENABLE_PORTBLAS_BACKEND=ON -DENABLE_MKLCPU_BACKEND=OFF -DTARGET_DOMAINS=blas -DCMAKE_CXX_FLAGS="-fsycl -fsycl-targets=spir64"
```

```suggestion
          options: -DENABLE_PORTBLAS_BACKEND=ON -DENABLE_MKLCPU_BACKEND=OFF -DENABLE_MKLGPU_BACKEND=OFF -DTARGET_DOMAINS=blas
```

The sycl flags shouldn't be needed, `-fsycl` is added automatically in oneMKL and `-fsycl-targets=spir64` is the default.
We should also disable the MKLGPU backend as it is not compatible the portBLAS one: https://github.com/oneapi-src/oneMKL/blob/develop/CMakeLists.txt#L115

---

I had to do this because it defaults to spirv+cuda: https://github.com/oneapi-src/oneMKL/actions/runs/9180847269/job/25246216751#step:6:128:~:text=per_kernel%20%2DWall%20%2DWextra%20%2DWshadow%20%2DWconversion%20%2DWpedantic%20%2Dfsycl%20%2Dfsycl%2Dunnamed%2Dlambda-,%2Dfsycl%2Dtargets%3Dnvptx64%2Dnvidia%2Dcuda%2Cspir64,-%2DMD%20%2DMT%20bin/dft/backends/portfft/CMakeFiles/onemkl_dft_portfft_obj.dir/descriptor

and the portfft recommended fixing it by setting `CMAKE_CXX_FLAGS`:
https://oneapi-src.github.io/oneMKL/building_the_project.html#building-for-portblas:~:text=The%20portFFT%20library%20is%20compiled%20using%20the%20same%20%2Dfsycl%2Dtargets

If you set CMAKE_CXX_FLAGS for the target, then you also need to set `-fsycl`

portblas talks about `PORTBLAS_TUNING_TARGET`:  https://oneapi-src.github.io/oneMKL/building_the_project.html#building-for-portblas:~:text=%2DDPORTBLAS_TUNING_TARGET%3D%3Ctarget%3E

It would be better if it defaulted to only `spirv` and the documentation was clearer about how to change the target.

What is the simplest way to get it to only target CPU?

---

For portBLAS the tuning target is optional. We don't need to set it for CPU or for quick tests. From my local tests the `CMAKE_CXX_FLAGS` can be removed.

I can see we have an issue with the portFFT backend. We wanted the default to also run on Nvidia but it seems we only tested it on machines where CUDA was installed so using `nvptx64-nvidia-cuda,spir64` by default worked. We'll need to improve this in the future.

---

Will remove for portblas